master/buildbot/www/hooks/github.py

```diff
@@ -73,7 +74,13 @@ def _get_payload(self, request):
             if hash_type != 'sha1':
                 raise ValueError('Unknown hash type: %s' % (hash_type,))
 
-            mac = hmac.new(self._secret, msg=content, digestmod=sha1)
+            _secret = self._secret
+            if isinstance(_secret, text_type):
```

Comment:
probably something like this:
```
def as_utf8_bytes(value):
    if isinstance(value, text_types):
        value = value.encode('utf-8')
    return value
```

one might add a check that otherwise it's of correct type.